File: OrangeJuice.Server.Core/Web/HttpUtility.cs
using System.Collections.Specialized;



namespace OrangeJuice.Server.Web

{

	public sealed class HttpUtility
	{

		public static NameValueCollection ParseQueryString(string query, IUrlEncoder urlEncoder)

		{

			var coll = System.Web.HttpUtility.ParseQueryString(query);
			return new QueryStringNameValueCollection(
coll, urlEncoder);

		}

	}

}

File: OrangeJuice.Server.Core.FSharp/Web/HttpUtility.fs
namespace OrangeJuice.Server.FSharp.Web

open System.Collections.Specialized

open OrangeJuice.Server.Web

type HttpUtility() =
    static member ParseQueryString(query : string, urlEncoder : IUrlEncoder) : NameValueCollection =
        let coll = System.Web.HttpUtility.ParseQueryString(query)
        new QueryStringNameValueCollection(coll, urlEncoder) :> NameValueCollection
